src/edu/csus/ecs/pc2/ui/ContestClockDisplay.java

```diff
@@ -512,37 +535,48 @@ protected void updateScheduledStartLabels() {
 
         if (schedAndRemainingTimeLabelList.size() > 0) {
 
-//            String hint = "Still no scheduled start";
-
             ContestTime contestTime = contest.getContestTime();
 
-            String text = getScheduleOrRemainingTime(contestTime);
+            String [] textAndHint = getScheduleOrRemainingTimeAndHint(contestTime);
 
-            for (JLabel jLabel : schedAndRemainingTimeLabelList) {
-                jLabel.setText(text);
-                jLabel.setToolTipText(text);
+            if(textAndHint != null && textAndHint.length == 2) {
+                for (JLabel jLabel : schedAndRemainingTimeLabelList) {
+                    jLabel.setText(textAndHint[0]);
+                    jLabel.setToolTipText(textAndHint[1]);
+                }
             }
         }
     }
 
-    public String getScheduleOrRemainingTime(ContestTime contestTime) {
-        
-        String text = NO_SCHEDULE_START_TIME; 
-        
-        boolean showRemainingTime = true;
-        
-        if (scheduledStartTime  != null && ! contestTime.isContestStarted()){
-            showRemainingTime = false;
-        }
-        
-        if (showRemainingTime){
-            
-            /**
-             * Show Remaining time count down. 
+    public String [] getScheduleOrRemainingTimeAndHint(ContestTime contestTime) {
```

Comment:
I feel like the name of this method should contain the word `Scheduled` (instead of `Schedule`).  As named, it sounds like it's getting either "a schedule [of something] or a Remaining time and hint" -- whereas it's intended to convey that it is getting either "a Scheduled _Time_ or a Remaining Time"...
 (And yes, I'm aware it was pre-existing.  )

Also, it would be nice if there was some minimal JavaDoc explaining what the method is doing and how the parameter is used, especially since it's returning different kinds of things depending on the state of the contest.
(And again, yes I'm aware this method, without any JavaDoc, was pre-existing..)

Reply:
Fixed in next push.